src/leo-client-app/src/components/EditSchedules.tsx

```diff
@@ -0,0 +1,243 @@
+
+
+import { sendCommandSchedule } from "@/constants/api";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import React, { useEffect, useState } from "react";
+import ViewScheduleCard from "./ViewScheduleCard";
+import { color } from 'd3';
+import '../components/EditSchedules.css'; // Your stylesheet filename
+
+
+
+
+const Scheduler: React.FC = () => {
+  const queryClient = useQueryClient();
+  const router = useRouter();
+
+
+  // TODO: Dynamicall get satelliteId from somewhere
+  const satelliteId = "655acd63d122507055d3d2ea";
+
+  const adminUserId: string = "65a5e11fe0d601e0e8c4a385";
+
+  // admin
+  // const userId: string = "65a5e11fe0d601e0e8c4a385";
+
+  // operator
+  const userId: string = "65a8181f36ea10b4366e1dd9";
+
+  const scheduleId = "65a8182036ea10b4366e1de6";
+  const isAdmin = adminUserId === userId;
+
+  const [validCommands, setValidCommands] = useState([]);
+  const [currentSchedule, setCurrentSchedule] = useState<string[]>([]);
+
+  const fetchValidCommands = (satelliteId: string) => {
+    if (isAdmin) {
+      fetch(
+        `http://localhost:3001/satellite/getSatellite?satelliteId=${satelliteId}`
+      )
+        .then((response) => response.json())
+        .then((data) => {
+          setValidCommands(data.satellite.validCommands);
+        })
+        .catch((error) => {
+          console.error("Error fetching valid commands:", error);
+        });
+    } else {
+      fetch(
+        `http://localhost:3001/satelliteUser/getCommandsBySatelliteAndUser?satelliteId=${satelliteId}&userId=${userId}`
+      )
+        .then((response) => response.json())
+        .then((data) => {
+          setValidCommands(data.record[0].validCommands);
+        })
+        .catch((error) => {
+          console.error("Error fetching valid commands:", error);
+        });
+    }
+  };
+
+  const addCommand = (command: string) => {
+    setCurrentSchedule((prevCommands) => [...prevCommands, command]);
+  };
+
+  const removeCommand = (index: number) => {
+    setCurrentSchedule((currentSchedule) =>
+      currentSchedule.filter((_, i) => i !== index)
+    );
+  };
+
+  useEffect(() => {
+    fetchValidCommands(satelliteId);
+  }, [satelliteId]);
+
+  // Mutation function
+  const { mutate } = useMutation({
+    mutationFn: () =>
+      sendCommandSchedule(userId, scheduleId, satelliteId, currentSchedule),
+
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ["useGetCommandsBySchedule"] });
+      setCurrentSchedule([]);
+    },
+
+    onSettled: () => {
+      queryClient.invalidateQueries({ queryKey: ["useGetCommandsBySchedule"] });
+      setCurrentSchedule([]);
+    },
+  });
+
+  // Function will load schedule somewhere, currently console log for POC demo
+  const sendSchedule = async () => {
+    mutate();
+  };
+
+  return (
+    <div
+      style={{
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "flex-start",
+        gap: "4rem",
+      }}>
```

Comment:
inline css should be injected as props - sx = {{}}, see other components for examples